ml-prims/src/linalg/svd.h

```diff
@@ -51,18 +53,18 @@ namespace LinAlg {
 template <typename T>
 void svdQR(T *in, int n_rows, int n_cols, T *sing_vals, T *left_sing_vecs,
            T *right_sing_vecs, bool trans_right, bool gen_left_vec, bool gen_right_vec,
-           cusolverDnHandle_t cusolverH, cublasHandle_t cublasH, cudaStream_t stream,
-           DeviceAllocator &mgr) {
+           cusolverDnHandle_t cusolverH, cublasHandle_t cublasH,
+           std::shared_ptr<deviceAllocator> allocator, cudaStream_t stream) {
```

Comment:
Could pass in `cumlHandle_impl` instead of cuBLAS handle, cuSolver handle and allocator.

Reply:
will be discussed in #498.